### ISSUE TITLE
Error when running the rebuild command.

### ISSUE THREAD

**Issue by rexon07** [tilmanmoser/bbb-video-download#41]
error:

 sudo -u bigbluebutton docker-compose run --rm --user 998:998 app node index.js -i /var/bigbluebutton/published/presentation/ed35d717c594a4dda2fb1861866c6d50f50dd605-1605167443263 -o /var/bigbluebutton/published/presentation/ed35d717c594a4dda2fb1861866c6d50f50dd605-1605167443263/video.mp4
WARNING: Error loading config file: /root/.docker/config.json: stat /root/.docker/config.json: permission denied
[Start] Rendering downloadable video for BBB presentation  {
  args: Namespace(input='/var/bigbluebutton/published/presentation/ed35d717c594a4dda2fb1861866c6d50f50dd605-1605167443263', output='/var/bigbluebutton/published/presentation/ed35d717c594a4dda2fb1861866c6d50f50dd605-1605167443263/video.mp4'),
  format: 'mp4',
  docker: true
}
[Error] Failed rendering downloable video for BBB presentation  {
  args: Namespace(input='/var/bigbluebutton/published/presentation/ed35d717c594a4dda2fb1861866c6d50f50dd605-1605167443263', output='/var/bigbluebutton/published/presentation/ed35d717c594a4dda2fb1861866c6d50f50dd605-1605167443263/video.mp4'),
  format: 'mp4',
  docker: true
} Error: EACCES: permission denied, mkdtemp './tmp/dataXXXXXX'
    at Object.mkdtempSync (fs.js:1869:3)
    at module.exports.createVideo (/home/bigbluebutton/modules/processor.js:13:25)
    at run (/home/bigbluebutton/index.js:10:15)
    at Object.<anonymous> (/home/bigbluebutton/index.js:20:1)
    at Module._compile (internal/modules/cjs/loader.js:1015:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1035:10)
    at Module.load (internal/modules/cjs/loader.js:879:32)
    at Function.Module._load (internal/modules/cjs/loader.js:724:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12)
    at internal/main/run_main_module.js:17:47 {
  errno: -13,
  syscall: 'mkdtemp',
  code: 'EACCES',
  path: './tmp/dataXXXXXX'
}

kindly let me know what am I missing here.
thanks

**Comment by tilmanmoser** [comment]
Check ownership and permissions of /opt/bbb-video-download/tmp. 
```bash
chown bigbluebutton:bigbluebutton /opt/bbb-video-download/tmp
chmod ug+rw /opt/bbb-video-download/tmp
```

**Comment by rexon07** [comment]
Hi @tilmanmoser  thanks for your reply it works now . But still get a warning  of permission error on config.json file can please let me the know the reason for this.

sudo -u bigbluebutton docker-compose run --rm --user 998:998 app node index.js -i /var/bigbluebutto                                                                                        n/published/presentation/ed35d717c594a4dda2fb1861866c6d50f50dd605-1605173835641                                                                                         -o /var/bigbluebutton/published/presentation/ed35d717c594a4dda2fb1861866c6d50f50                                                                                        dd605-1605173835641/video.mp4
WARNING: Error loading config file: /root/.docker/config.json: stat /root/.docke                                                                                        r/config.json: permission denied
[Start] Rendering downloadable video for BBB presentation  {
  args: Namespace(input='/var/bigbluebutton/published/presentation/ed35d717c594a                                                                                        4dda2fb1861866c6d50f50dd605-1605173835641', output='/var/bigbluebutton/published                                                                                        /presentation/ed35d717c594a4dda2fb1861866c6d50f50dd605-1605173835641/video.mp4')                                                                                        ,
  format: 'mp4',
  docker: true
}
[End] Finished rendering downloadable video for BBB presentation in 98.056 seconds  {
  args: Namespace(input='/var/bigbluebutton/published/presentation/ed35d717c594a4dda2fb1861866c6d50f50dd605-1605173835641', output='/var/bigbluebutton/published/presentation/ed35d717c594a4dda2fb1861866c6d50f50dd605-1605173835641/video.mp4'),
  format: 'mp4',
  docker: true,
  workdir: './tmp/dataNDfjnJ'
}

thanks.

**Comment by tilmanmoser** [comment]
That‘s fine. The script doesn‘t need to read that file (which belongs to root). It only checks for existence to determine whether it‘s running in a docker environment. If that‘s the case it‘s save to use linux commands to copy files. Otherwise it would have to use js commands, which are translated to os native operations. But this might lead to other issues in docker env, when the file is moved cross mounted volumes. So - short: it dies what it should ;-)

**Comment by rexon07** [comment]
thanks got it!! :)

**Comment by MOPO3KO** [comment]
Hey.
I am using an old version of BBB, the update is complicated by a number of reasons that require a fight, but the main one is the lack of the Internet on the server (it is forbidden to enable it).
The current script, when running, gives an error:

bbb@ubuntu:/opt/bbb-video-download$ sudo -u bigbluebutton docker-compose run --rm --user 997:997 app node index.js -i /var/bigbluebutton/published/presentation/5941b4e7060e134bb7389947897204258fdf5a11-1603880721595 -o /var/bigbluebutton/published/presentation/5941b4e7060e134bb7389947897204258fdf5a11-1603880721595/video.mp4
[Start] Rendering downloadable video for BBB presentation  {
  args: Namespace(input='/var/bigbluebutton/published/presentation/5941b4e7060e134bb7389947897204258fdf5a11-1603880721595', output='/var/bigbluebutton/published/presentation/5941b4e7060e134bb7389947897204258fdf5a11-1603880721595/video.mp4'),
  format: 'mp4',
  docker: true
}
mv: can't create '/var/bigbluebutton/published/presentation/5941b4e7060e134bb7389947897204258fdf5a11-1603880721595/video.mp4': Permission denied
[Error] Failed rendering downloable video for BBB presentation  {
  args: Namespace(input='/var/bigbluebutton/published/presentation/5941b4e7060e134bb7389947897204258fdf5a11-1603880721595', output='/var/bigbluebutton/published/presentation/5941b4e7060e134bb7389947897204258fdf5a11-1603880721595/video.mp4'),
  format: 'mp4',
  docker: true,
  workdir: './tmp/dataBnGdDN'
} Error: Command failed: mv ./tmp/dataBnGdDN/video.mp4 /var/bigbluebutton/published/presentation/5941b4e7060e134bb7389947897204258fdf5a11-1603880721595/video.mp4
mv: can't create '/var/bigbluebutton/published/presentation/5941b4e7060e134bb7389947897204258fdf5a11-1603880721595/video.mp4': Permission denied

    at checkExecSyncError (child_process.js:635:11)
    at Object.execSync (child_process.js:671:15)
    at moveFile (/home/bigbluebutton/modules/processor.js:101:22)
    at module.exports.createVideo (/home/bigbluebutton/modules/processor.js:34:9)
    at async run (/home/bigbluebutton/index.js:10:9) {
  status: 1,
  signal: null,
  output: [
    null,
    <Buffer >,
    <Buffer 6d 76 3a 20 63 61 6e 27 74 20 63 72 65 61 74 65 20 27 2f 76 61 72 2f 62 69 67 62 6c 75 65 62 75 74 74 6f 6e 2f 70 75 62 6c 69 73 68 65 64 2f 70 72 65 ... 95 more bytes>
  ],
  pid: 259,
  stdout: <Buffer >,
  stderr: <Buffer 6d 76 3a 20 63 61 6e 27 74 20 63 72 65 61 74 65 20 27 2f 76 61 72 2f 62 69 67 62 6c 75 65 62 75 74 74 6f 6e 2f 70 75 62 6c 69 73 68 65 64 2f 70 72 65 ... 95 more bytes>
}

at command output:
`cat /etc/passwd | grep bigbluebutton`

получаю:
`bigbluebutton:x:997:997:bbb-apps-akka user-daemon:/home/bigbluebutton:/bin/false`

of course I am aiming to update the version, but it is not up to me ...

please, tell me how to get out of the situation at the moment?
